tools/db_bench_tool.cc

```diff
@@ -8185,6 +8207,50 @@ class Benchmark {
     }
   }
 
+  void Backup(ThreadState* thread) {
+    DB* db = SelectDB(thread);
+    std::string backup_dir = FLAGS_db + "/backups";
+    std::unique_ptr<BackupEngineOptions> engine_options_(
+        new BackupEngineOptions(backup_dir));
+    Status s;
+    BackupEngine* backup_engine;
+    if (FLAGS_backup_rate_limit > 0) {
+      engine_options_.get()->backup_rate_limiter.reset(NewGenericRateLimiter(
+          FLAGS_backup_rate_limit, 100000 /* refill_period_us */,
+          10 /* fairness */, RateLimiter::Mode::kAllIo));
+    }
+    // Build new backup of the entire DB
+    engine_options_->destroy_old_data = true;
+    s = BackupEngine::Open(FLAGS_env, *engine_options_, &backup_engine);
+    assert(s.ok());
+    s = backup_engine->CreateNewBackup(db);
+    assert(s.ok());
+    std::vector<BackupInfo> backup_info;
+    backup_engine->GetBackupInfo(&backup_info);
+    // Verify that a new backup is created
+    assert(backup_info.size() == 1);
+  }
+
+  void Restore(ThreadState* thread) {
+    std::string backup_dir = FLAGS_db + "/backups";
+    std::unique_ptr<BackupEngineOptions> engine_options_(
+        new BackupEngineOptions(backup_dir));
+    if (FLAGS_restore_rate_limit > 0) {
+      engine_options_.get()->restore_rate_limiter.reset(NewGenericRateLimiter(
+          FLAGS_restore_rate_limit, 100000 /* refill_period_us */,
+          10 /* fairness */, RateLimiter::Mode::kAllIo));
+    }
+    BackupEngineReadOnly* backup_engine;
+    Status s = BackupEngineReadOnly::Open(Env::Default(), *engine_options_,
+                                          &backup_engine);
+    assert(s.ok());
+    std::string restore_dir =
+        FLAGS_db + "/restore_" + std::to_string(thread->rand.Next());
```

Comment:
Any particular reason to add some randomness (i.e, thread->rand.Next())  to the restore folder name?